### ISSUE TITLE
`multipass mount` while cloud-init is still running apt commands fails

### ISSUE THREAD

**Issue by OddBloke** [canonical/multipass#832]
With this cloud-config (ubuntu-dev-tools has a bunch of dependencies, so it gives us a decent window to reproduce in):

```
#cloud-config
packages: [ubuntu-dev-tools]
```

I see the following:

```
$ multipass launch xenial --cloud-init cloud-config
Launched: convincing-hermit

$ multipass mount -v my-dir convincing-hermit:/tmp/my-dir
[2019-06-10T15:28:14.164] [warning] [sshfs mount] Unable to determine if 'sshfs' is installed: 

Enabling support for mounting \[2019-06-10T15:28:16.428] [warning] [daemon] Failed to install 'sshfs', error message: '
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/'

 \[2019-06-10T15:28:16.639] [warning] [daemon] Failed to install 'sshfs', error message: '
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/'

 /[2019-06-10T15:28:18.313] [warning] [daemon] Failed to install 'sshfs', error message: '
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Could not get lock /var/lib/apt/lists/lock - open (11: Resource temporarily unavailable)
E: Unable to lock directory /var/lib/apt/lists/'

mount failed: Error enabling mount support in 'convincing-hermit'

**Comment by townsend2010** [comment]
@OddBloke,

Thanks for reporting this.  We kind have fleshed this out in #696, so I'll just close this as a dup of that.

**Comment by townsend2010** [comment]
I'm reopening this in that `mount` should wait for the apt lock to become free before bailing out unless a reasonable amount of time has passed, ie, a stale lock is left behind from something else.

**Comment by Saviq** [comment]
This will get solved by #627.
